From 4b6863bb635a8ac568a3c44f77d1ee388593a0fd Mon Sep 17 00:00:00 2001
From: Momo <mynameisjoyg@gmail.com>
Date: Fri, 9 Oct 2026 12:59:05 +0800
Subject: [PATCH] Add ads in AccountActivity.

	modified:   app/src/main/java/com/joyg/quizhero/AccountActivity.kt
	modified:   app/src/main/res/layout/activity_account.xml
---
 .../java/com/joyg/quizhero/AccountActivity.kt   | 17 +++++++++++++++++
 app/src/main/res/layout/activity_account.xml    | 17 +++++++++++++++++
 2 files changed, 34 insertions(+)

diff --git a/app/src/main/java/com/joyg/quizhero/AccountActivity.kt b/app/src/main/java/com/joyg/quizhero/AccountActivity.kt
index a64f433..133c592 100644
--- a/app/src/main/java/com/joyg/quizhero/AccountActivity.kt
+++ b/app/src/main/java/com/joyg/quizhero/AccountActivity.kt
@@ -3,7 +3,12 @@ package com.joyg.quizhero
 import android.os.Bundle
 import android.widget.TextView
 import androidx.activity.ComponentActivity
+import com.google.android.gms.ads.AdRequest
+import com.google.android.gms.ads.AdSize
+import com.google.android.gms.ads.MobileAds
 
+import com.google.android.gms.ads.AdView
+private lateinit var adView: AdView
 class AccountActivity : ComponentActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
@@ -24,5 +29,17 @@ class AccountActivity : ComponentActivity() {
         tvUserId.text = "User ID: $userId"
         tvTotalCorrectCount.text = "總答對題數: $totalCorrectCount"
         tvTotalWrongCount.text = "總答錯題數: $totalWrongCount"
+
+
+        adView = findViewById<AdView>(R.id.adView)
+
+        // 3. Banner廣告
+        // 初始化 Google Mobile Ads SDK
+        MobileAds.initialize(this) {}
+
+        // 綁定並載入廣告
+        adView = findViewById(R.id.adView)
+        val adRequest = AdRequest.Builder().build()
+        adView.loadAd(adRequest)
     }
 }
diff --git a/app/src/main/res/layout/activity_account.xml b/app/src/main/res/layout/activity_account.xml
index 3d048b0..17868a1 100644
--- a/app/src/main/res/layout/activity_account.xml
+++ b/app/src/main/res/layout/activity_account.xml
@@ -72,4 +72,21 @@
         </LinearLayout>
     </com.google.android.material.card.MaterialCardView>
 
+    <!-- 加入一個彈性伸展空間，將 AdView 推至最底部 -->
+    <Space
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        android:layout_weight="1" />
+
+    <!-- 置底的 AdMob Banner 廣告 -->
+    <com.google.android.gms.ads.AdView
+        android:id="@+id/adView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_horizontal"
+        android:layout_marginTop="8dp"
+        android:layout_marginBottom="16dp"
+        app:adSize="BANNER"
+        app:adUnitId="@string/bannerAdUnitId" />
+
 </LinearLayout>
-- 
2.55.0.windows.5

